### PR TITLE
Deal with some clippy warnings

### PR DESCRIPTION
**Pull request by thomaseizinger**
Can drop individual commits if some of the changes are not wanted.

I purposely left everything in `relay` untouched to not interfere with https://github.com/libp2p/rust-libp2p/pull/2059 too much.

### CODE REVIEW

File: muxers/mplex/src/codec.rs [hunk 1]
```diff
@@ -244,25 +244,25 @@ impl Encoder for Codec {
     fn encode(&mut self, item: Self::Item, dst: &mut BytesMut) -> Result<(), Self::Error> {
         let (header, data) = match item {
             Frame::Open { stream_id } => {
-                (u64::from(stream_id.num) << 3, Bytes::new())
+                (stream_id.num << 3, Bytes::new())
```
Comment (mxinden) [inline muxers/mplex/src/codec.rs:247]: Thanks! I should have caught this in https://github.com/libp2p/rust-libp2p/pull/2094/.

**Review by mxinden — APPROVED** [review]
Thank you @thomaseizinger!

> I purposely left everything in relay untouched to not interfere with #2059 too much.

Very considerate. Thank you.

**Comment by mxinden** [comment]
As an aside: I would as well be fine failing CI on clippy _warning_ level lints.

**Comment by thomaseizinger** [comment]
> As an aside: I would as well be fine failing CI on clippy _warning_ level lints.

I would like that! If we do that now, it will interfere with your relay changes though. Let me know if that is okay for you.

**Comment by mxinden** [comment]
> > As an aside: I would as well be fine failing CI on clippy _warning_ level lints.
> 
> I would like that! If we do that now, it will interfere with your relay changes though. Let me know if that is okay for you.

That would be fine by me. 

Me being slow at tackling https://github.com/libp2p/rust-libp2p/issues/2052 should not slow down all of you.

**Comment by thomaseizinger** [comment]
I opened https://github.com/libp2p/rust-libp2p/pull/2148.

**Pull request merged**